src/d/actor/d_a_warpfout.cpp

```diff
@@ -4,125 +4,236 @@
+    &daWarpfout_c::initWarp4, &daWarpfout_c::initEnd,
+};
+
+static daWarpfout_c::EventActionFunc event_action_tbl[] = {
+    &daWarpfout_c::actWarp1, &daWarpfout_c::actWarp2, &daWarpfout_c::actWarp3,
+    &daWarpfout_c::actWarp4, &daWarpfout_c::actEnd,
+};
+
+static char* action_table[] = {
+    "WARP1", "WARP2", "WARP3", "WARP4", "END",
+};
 
 /* 00000078-00000080       .text _delete__12daWarpfout_cFv */
-BOOL daWarpfout_c::_delete() {
-    /* Nonmatching */
+bool daWarpfout_c::_delete() {
+    return true;
 }
 
 /* 00000080-0000008C       .text CreateInit__12daWarpfout_cFv */
 void daWarpfout_c::CreateInit() {
-    /* Nonmatching */
+    unknownInitFlag = -0x1;
 }
 
 /* 0000008C-000000E4       .text _create__12daWarpfout_cFv */
 s32 daWarpfout_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daWarpfout_c);
+    CreateInit();
+
+    return cPhs_COMPLEATE_e;
 }
 
 /* 000000E4-00000124       .text _execute__12daWarpfout_cFv */
-BOOL daWarpfout_c::_execute() {
-    /* Nonmatching */
+bool daWarpfout_c::_execute() {
+    checkOrder();
+    demo_proc();
+    eventOrder();
+    return true;
 }
 
 /* 00000124-00000128       .text checkOrder__12daWarpfout_cFv */
-void daWarpfout_c::checkOrder() {
-    /* Nonmatching */
-}
+void daWarpfout_c::checkOrder() {}
 
 /* 00000128-0000012C       .text eventOrder__12daWarpfout_cFv */
-void daWarpfout_c::eventOrder() {
-    /* Nonmatching */
-}
+void daWarpfout_c::eventOrder() {}
 
 /* 0000012C-00000248       .text demo_proc__12daWarpfout_cFv */
 void daWarpfout_c::demo_proc() {
-    /* Nonmatching */
+    dComIfG_inf_c* gameinfo = &g_dComIfG_gameInfo;
+
+    mStaffId = dComIfGp_evmng_getMyStaffId("Warpfo");
+
+    if (gameinfo->play.getEvent().getMode() != dEvtMode_NONE_e && !mEvtInfo.checkCommandTalk() &&
+        mStaffId != -1)
+    {
+        int actIdx =
+            dComIfGp_evmng_getMyActIdx(mStaffId, action_table, ARRAY_SIZE(action_table), 0, 0);
+
+        if (actIdx == -1) {
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+
+        if (dComIfGp_evmng_getIsAddvance(mStaffId)) {
+            (this->*event_init_tbl[actIdx])(mStaffId);
+        }
+        if ((this->*event_action_tbl[actIdx])(mStaffId)) {
+            dComIfGp_evmng_cutEnd(mStaffId);
+        }
+    }
 }
 
 /* 00000248-000002A8       .text initWarp1__12daWarpfout_cFi */
-void daWarpfout_c::initWarp1(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::initWarp1(int) {
+    mTimer = 5;
+    JAIZelBasic::zel_basic->seStart(JA_SE_LK_BOSS_WARP_EFF_ED, NULL, 0, 0, 1.0, 1.0, -1.0, -1.0, 0);
 }
 
 /* 000002A8-00000344       .text actWarp1__12daWarpfout_cFi */
-void daWarpfout_c::actWarp1(int) {
-    /* Nonmatching */
+BOOL daWarpfout_c::actWarp1(int) {
+    if (mTimer == 0x4) {
+        cXyz effectPos = dComIfGp_getLinkPlayer()->current.pos;
+        effectPos.y += 40.0f;
+
+        set_effect_wind01(effectPos, 0);
+    }
+
+    if (!cLib_calcTimer(&mTimer)) {
```

Comment:
```suggestion
    if (cLib_calcTimer(&mTimer) == 0) {
```
not a big deal but calctimer returns the value remaining on the timer, not a boolean